components/store_state.py

```diff
@@ -14,7 +16,7 @@ def get_store_component(self, store_id):
     @staticmethod
     def save_state_to_store(store_data, key, value):
         """Save data to dcc.Store."""
-        if store_data is None:
+        if store_data is None or not isinstance(store_data, dict):
             store_data = {}
         store_data[key] = value
```

Comment:
are the keys the input data? If so please make sure we use the same id as in the main application.